### PR TITLE
CredxAnoncreds verifier functionality support

### PR DESCRIPTION
**Pull request by gmulhearn**
#699 Implements credx verifier method and tests

* Add new error `ProofRejected` to map to credx and vdrtools errors
* minor fix to logging in setupprofile
* changed tests as possible to use a dynamic profile (not locked to indy) for verifier functionality
* ^ this meant using `holder_setup` profile, after the work for credx issuer APIs are complete, ALL tests should be able to use `SetupProfile::run` rather than `SetupProfile::run_indy`

### CODE REVIEW

File: libvcx/src/errors/mapping_from_ariesvcx.rs [hunk 1]
```diff
@@ -79,6 +79,7 @@ impl From<AriesVcxErrorKind> for LibvcxErrorKind {
             AriesVcxErrorKind::VdrToolsError(num) => LibvcxErrorKind::LibndyError(num),
             AriesVcxErrorKind::NoAgentInformation => LibvcxErrorKind::NoAgentInformation,
             AriesVcxErrorKind::InvalidMessageFormat => LibvcxErrorKind::InvalidMessageFormat,
+            AriesVcxErrorKind::ProofRejected => LibvcxErrorKind::LibndyError(405), // AnoncredsProofRejected Vdrtools error code
```
Comment (Patrik-Stas) [inline libvcx/src/errors/mapping_from_ariesvcx.rs:82]: Let's also create new `LibvcxErrorKind::ProofRejected`

File: aries_vcx/src/common/proofs/verifier/verifier.rs [hunk 2]
```diff
@@ -195,17 +193,17 @@ pub mod unit_tests {
                 .await
                 .unwrap();
             assert_eq!(
-                validate_indy_proof(&setup.profile, &prover_proof_json, &proof_req_json)
+                validate_indy_proof(&holder_setup.profile, &prover_proof_json, &proof_req_json)
                     .await
                     .unwrap_err()
                     .kind(),
-                AriesVcxErrorKind::VdrToolsError(405)
+                AriesVcxErrorKind::ProofRejected
```
Comment (Patrik-Stas) [inline aries_vcx/src/common/proofs/verifier/verifier.rs:200]: 👍  just run into this as well while improving err handling a bit in https://github.com/hyperledger/aries-vcx/pull/721 and though you might have as well by what you mentioned on the last call - and yeah here it is :-)

Comment (Patrik-Stas) [reply aries_vcx/src/common/proofs/verifier/verifier.rs:200]: After this PR and #721, there won't be really explicit usage of `VdrToolsError` in aries-vcx or vdrtools - just will be left as variant for vdrtools errors which don't really need to be mapped to something (bunch of which should probably be removed)

Comment (gmulhearn) [reply aries_vcx/src/common/proofs/verifier/verifier.rs:200]: Thanks! reviewing 721 now, let's get that in first and I'll rebase and finish up this MR

**Review by Patrik-Stas — APPROVED** [review]

**Review by mirgee — APPROVED** [review]

**Pull request merged**